### PR TITLE
The Azure ServiceBus Queue -name must match the Dapr pubsub topic

### PR DESCRIPTION
**Pull request by TomasEkeli**
## Description

Describe how to get past some stumbling-blocks when using Azure ServiceBus Queue as a pubsub component.

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md [hunk 1]
```diff
@@ -181,6 +181,17 @@ When subscribing to a topic, you can configure `bulkSubscribe` options. Refer to
...
+If you are using a Shared access policy to connect to the queue that policy must have the ability to "Manage" the queue. To work with a dead-letter queue the policy must live on the Service Bus Namespace that contains both the main queue and the dead-letter queue.
+{{% /alert %}}
+
+### Retry policy and dead-letter queues
+
+An Azure ServiceBus Queue has a dead-letter queue by default. By default the messages are retried the "Max delivery count" number of times (defaults to 10, can be set up to 2000). These retries happen very rapidly and the message is put in the dead-letter queue if no success is returned.
+
+Dapr Pubsub has its own dead-letter queue -concept you can use instead. This lets you control the retry-policy and you can subscribe to the dead-letter queue through Dapr. Set up a separate queue as that dead-letter queue in the Azure Service Bus Namespace, and a resilience component that defines how to retry. You can subscribe to that (topic) to get the failed messages and deal with them. E.g. setting up a dead-letter queue `orders-dlq` in the subscription and a resilience lets your subscribe to the topic `orders-dlq` to handle failed messages.
```
Comment (yaron2) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md]: Better to link to the Dapr dead letter topic doc here. Replace `resilience component` with `resilience policy` please

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md [hunk 2]
```diff
@@ -181,6 +181,19 @@ When subscribing to a topic, you can configure `bulkSubscribe` options. Refer to
 
 Follow the instructions [here](https://learn.microsoft.com/azure/service-bus-messaging/service-bus-quickstart-portal) on setting up Azure Service Bus Queues.
 
+{{% alert title="Note" color="primary" %}}
+Your queue must have the same name as the Topic you are publishing to with Dapr. E.g. if you are publishing to the pubsb "myPubsub" on the topic "orders" your queue must be named "orders".
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md]:
```suggestion
Your queue must have the same name as the topic you are publishing to with Dapr. For example, if you are publishing to the pub/sub `"myPubsub"` on the topic `"orders"`, your queue must be named `"orders"`.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md [hunk 3]
```diff
@@ -181,6 +181,19 @@ When subscribing to a topic, you can configure `bulkSubscribe` options. Refer to
...
+If you are using a Shared access policy to connect to the queue that policy must have the ability to "Manage" the queue. To work with a dead-letter queue the policy must live on the Service Bus Namespace that contains both the main queue and the dead-letter queue.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md]:
```suggestion
If you are using a shared access policy to connect to the queue, that policy must be able to "manage" the queue. To work with a dead-letter queue, the policy must live on the Service Bus Namespace that contains both the main queue and the dead-letter queue.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md [hunk 4]
```diff
@@ -181,6 +181,19 @@ When subscribing to a topic, you can configure `bulkSubscribe` options. Refer to
...
+{{% /alert %}}
+
+### Retry policy and dead-letter queues
+
+An Azure ServiceBus Queue has a dead-letter queue by default. By default the messages are retried the "Max delivery count" number of times (defaults to 10, can be set up to 2000). These retries happen very rapidly and the message is put in the dead-letter queue if no success is returned.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md]:
```suggestion
By default, an Azure Service Bus Queue has a dead-letter queue. The messages are retried the amount given for `maxDeliveryCount`. The default `maxDeliveryCount` value defaults to 10, but can be set up to 2000. These retries happen very rapidly and the message is put in the dead-letter queue if no success is returned.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md [hunk 5]
```diff
@@ -181,6 +181,19 @@ When subscribing to a topic, you can configure `bulkSubscribe` options. Refer to
...
+
+Dapr Pubsub has its own dead-letter queue -concept you can use instead. This lets you control the retry-policy and you can subscribe to the dead-letter queue through Dapr. Set up a separate queue as that dead-letter queue in the Azure Service Bus Namespace, and a resilience policy that defines how to retry. You can subscribe to that (topic) to get the failed messages and deal with them. E.g. setting up a dead-letter queue `orders-dlq` in the subscription and a resilience lets your subscribe to the topic `orders-dlq` to handle failed messages.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md]:
```suggestion
Dapr Pub/sub offers its own dead-letter queue concept that lets you control the retry policy and subscribe to the dead-letter queue through Dapr. 
1. Set up a separate queue as that dead-letter queue in the Azure Service Bus namespace, and a resilience policy that defines how to retry. 
1. Subscribe to the topic to get the failed messages and deal with them. 

For example, setting up a dead-letter queue `orders-dlq` in the subscription and a resiliency policy lets you subscribe to the topic `orders-dlq` to handle failed messages.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick review

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md (hunk 5 above)
Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md]: @TomasEkeli  - Dapr has "dead-letter topic concept" not "dead-letter queue concept". Need to change to dead-letter topic in these paragraphs. I presume that is what you are referring to. Can you confirm?

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md (hunk 2 above)
Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-servicebus-queues.md]: "Topic" should be "topic" (lowercase)

**Comment by msfussell** [comment]
@TomasEkeli Ping on looking at these review comments. Thanks

**Comment by msfussell** [comment]
@TomasEkeli - Ping again on this PR.

**Comment by msfussell** [comment]
@TomasEkeli - Another ping again on this PR.

**Comment by TomasEkeli** [comment]
sorry, been very busy. I have looked at these suggested changes, and they look excellent! thanks!

**Review by hhunter-ms — APPROVED** [review]
lgtm - may still need a final review from @yaron2

**Review by yaron2 — APPROVED** [review]

**Comment by marcduiker** [comment]
@holopin-bot @TomasEkeli  Thank you!

**Pull request merged**